examples/kokkos_assembly/kokkos_assembly.cpp

```diff
@@ -0,0 +1,241 @@
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <iostream>
+#include <map>
+#include <string>
+
+
+#include <omp.h>
+#include <Kokkos_Core.hpp>
+#include <ginkgo/ginkgo.hpp>
+
+
+// Creates a stencil matrix in CSR format for the given number of discretization
+// points.
+template <typename ValueType, typename IndexType>
+void generate_stencil_matrix(gko::matrix::Csr<ValueType, IndexType>* matrix)
+{
+    auto exec = matrix->get_executor();
+    const auto discretization_points = matrix->get_size()[0];
+
+    // Over-allocate storage for the matrix elements. Each row has 3 entries,
+    // except for the first and last one. To handle each row uniformly, we
+    // allocate space for 3x the number of rows.
+    gko::device_matrix_data<ValueType, IndexType> md(exec, matrix->get_size(),
+                                                     discretization_points * 3);
+
+    // Create Kokkos views on Ginkgo data.
+    Kokkos::View<IndexType*> v_row_idxs(md.get_row_idxs(), md.get_num_elems());
+    Kokkos::View<IndexType*> v_col_idxs(md.get_col_idxs(), md.get_num_elems());
+    Kokkos::View<ValueType*> v_values(md.get_values(), md.get_num_elems());
+
+    // Create the matrix entries. This also creates zero entries for the
+    // first and second row to handle all rows uniformly.
+    Kokkos::parallel_for(
+        "generate_stencil_matrix", md.get_num_elems(), KOKKOS_LAMBDA(int i) {
+            const ValueType coefs[] = {-1, 2, -1};
+            auto ofs = static_cast<IndexType>((i % 3) - 1);
+            auto row = static_cast<IndexType>(i / 3);
+            auto col = row + ofs;
+
+            // To prevent branching, a mask is used to set the entry to
+            // zero, if the column is out-of-bounds
+            auto mask =
+                static_cast<IndexType>(0 <= col && col < discretization_points);
+
+            v_row_idxs[i] = mask * static_cast<IndexType>(row);
+            v_col_idxs[i] = mask * static_cast<IndexType>(col);
```

Comment:
static_cast's are not really needed here.
```suggestion
            v_row_idxs[i] = mask * row;
            v_col_idxs[i] = mask * col;
```